src/main/java/org/apache/mesos/hdfs/config/HdfsFrameworkConfig.java

```diff
@@ -102,34 +144,40 @@ public double getZkfcCpus() {
   }
 
   public double getNameNodeCpus() {
-    return getConf().getDouble("mesos.hdfs.namenode.cpus", 1);
+    return getConf().getDouble("mesos.hdfs.namenode.cpus", NAMENODE_CPUS);
   }
 
   public double getJournalNodeCpus() {
-    return getConf().getDouble("mesos.hdfs.journalnode.cpus", 1);
+    return getConf().getDouble("mesos.hdfs.journalnode.cpus", JOURNAL_CPUS);
   }
 
   public double getDataNodeCpus() {
-    return getConf().getDouble("mesos.hdfs.datanode.cpus", 1);
+    return getConf().getDouble("mesos.hdfs.datanode.cpus", DATANODE_CPUS);
   }
 
   public double getTaskCpus(String taskName) {
+    double cpus = DEFAULT_CPUS;
     switch (taskName) {
-      case "zkfc" :
-        return getZkfcCpus();
-      case "namenode" :
-        return getNameNodeCpus();
-      case "datanode" :
-        return getDataNodeCpus();
-      case "journalnode" :
-        return getJournalNodeCpus();
+      case HDFSConstants.ZKFC_NODE_ID :
+        cpus = getZkfcCpus();
+        break;
+      case HDFSConstants.NAME_NODE_ID :
+        cpus = getNameNodeCpus();
+        break;
+      case HDFSConstants.DATA_NODE_ID :
+        cpus = getDataNodeCpus();
+        break;
+      case HDFSConstants.JOURNAL_NODE_ID :
+        cpus = getJournalNodeCpus();
+        break;
       default :
-        throw new RuntimeException("Invalid taskName=" + taskName);
+        log.error("Invalid taskName=" + taskName);
```

Comment:
Error and return DEFAULT_CPUS? Seems like a nasty bug in waiting. An invalid taskName/Type shouldn't be able to launch a task, even with some "reasonable" default. Let's throw an exception or somehow error to the caller.